src/core/interface.scala

```diff
@@ -33,6 +34,28 @@ object CaseClass:
     override def toString: String = s"Param($label)"
 
   object Param:
+
+    def apply[F[_], T, P](
+        name: String,
+        idx: Int,
+        repeated: Boolean,
+        cbn: CallByNeed[F[P]],
+        defaultVal: CallByNeed[Option[P]],
+        annotations: List[Any],
```

Comment:
does this only differ in using `List` instead of `IArray`? If so, can we keep a single variant? that way the public api (the one for which we need to keep compatibility) will be smaller

Reply:
Still stuck with it. I have issues of mapping from `Expr[List[Any]]` to `Expr[IArray[Any]]`.